spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/test/java/io/awspring/cloud/s3/S3TemplateIntegrationTests.java

```diff
@@ -183,6 +200,45 @@ void downloadsFile() throws IOException {
 		}
 	}
 
+	@Test
+	void createsWorkingSignedGetURL() throws IOException {
+		client.putObject(r -> r.bucket(BUCKET_NAME).key("file.txt"), RequestBody.fromString("hello"));
+		URL signedGetUrl = s3Template.createSignedGetURL(BUCKET_NAME, "file.txt", Duration.ofMinutes(1));
+
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		HttpGet httpGet = new HttpGet(signedGetUrl.toString());
+		HttpResponse response = httpClient.execute(httpGet);
+
+		try (InputStream content = response.getEntity().getContent()) {
+			String result = StreamUtils.copyToString(content, StandardCharsets.UTF_8);
+			assertThat(result).isEqualTo("hello");
+		}
+	}
+
+	@Test
+	void createsWorkingSignedPutURL() throws IOException {
+		ObjectMetadata metadata = ObjectMetadata.builder().metadata("testkey", "testvalue").build();
+		URL signedPutUrl = s3Template.createSignedPutURL(BUCKET_NAME, "file.txt", Duration.ofMinutes(1), metadata, "text/plain");
+
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		HttpPut httpPut = new HttpPut(signedPutUrl.toString());
+		httpPut.setHeader("x-amz-meta-testkey", "testvalue");
+		httpPut.setHeader("Content-Type","text/plain");
+		HttpEntity body = new StringEntity("hello");
+		httpPut.setEntity(body);
+
+		HttpResponse response = httpClient.execute(httpPut);
+		httpClient.close();
+
+		HeadObjectResponse headObjectResponse = client.headObject(HeadObjectRequest.builder()
```

Comment:
Genuine question - shouldn't we upload something in this test?

Reply:
I guess it technically isn't uploading a local file, but it is adding a new file "file.txt" to the bucket with the request body as its contents and then checking to make sure it is there at the end. Personally, I think knowing this works is enough to be confident that the signed URL itself works. The difference is just about how you use it. That's just my interpretation, though. I'm open to other opinions!